### PR TITLE
Add requirement edx-api-doc-tools and drop support for Ironwood

### PR DESCRIPTION
**Pull request by magajh**
This PR adds the latest version of edx-api-doc-tools, this requirement is needed for #125 
We are including edx-api-doc-tools 1.4.0 (latest version) since it has two new important features for the documentation:
- Option to include a body parameter in requests.
- Option to specify url patterns for generated docs.

However, this edx-api-doc-tools version has problems with the requirements for ironwood, meaning eox-core will no longer be backward compatible.

**Changes**
- Upgrade requirements 
- Delete settings for the tests with django1.11 and django2.2
- Delete requirements files: Django and Django22
- Add tests for python3.8 and  Debian versions: stretch and buster
- Since Pylint got upgraded to 2.5.0 (older versions do not support python3.8), some new code-style errors appeared. A list with some of the errors that were Disabled/Fixed is included next.
  - Disabled:
    - [import-outside-toplevel](https://github.com/PyCQA/pylint/issues/3067)
    - import-error for django_filters
  - Fixed:
    - [useless-object-inheritance](https://github.com/PyCQA/pylint/issues/2177)
    - [unnecessary-pass](https://github.com/PyCQA/pylint/issues/2208)

### CODE REVIEW

File: eox_core/api/v1/views.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 LOG = logging.getLogger(__name__)
 
 
-class UserQueryMixin(object):
+class UserQueryMixin():
```
Comment (morenol) [inline eox_core/api/v1/views.py:42]:
```suggestion
class UserQueryMixin:
```

File: eox_core/edxapp_wrapper/backends/users_h_v1.py [hunk 2]
```diff
@@ -175,12 +175,12 @@ def get_course_team_user(*args, **kwargs):
...
     """
     if settings.SERVICE_VARIANT == 'cms':
-        from contentstore.views.user import _course_team_user  # pylint: disable=import-error
+        from contentstore.views.user import _course_team_user  # pylint: disable=import-error,import-outside-toplevel
         return _course_team_user(*args, **kwargs)
     return None
 
 
-class FetchUserSiteSources(object):
+class FetchUserSiteSources():
```
Comment (morenol) [inline eox_core/edxapp_wrapper/backends/users_h_v1.py:183]:
```suggestion
class FetchUserSiteSources:
```

File: eox_core/edxapp_wrapper/backends/users_j_v1.py [hunk 3]
```diff
@@ -187,12 +187,12 @@ def get_course_team_user(*args, **kwargs):
...
     """
     if settings.SERVICE_VARIANT == 'cms':
-        from contentstore.views.user import _course_team_user  # pylint: disable=import-error
+        from contentstore.views.user import _course_team_user  # pylint: disable=import-error,import-outside-toplevel
         return _course_team_user(*args, **kwargs)
     return None
 
 
-class FetchUserSiteSources(object):
+class FetchUserSiteSources():
```
Comment (morenol) [inline eox_core/edxapp_wrapper/backends/users_j_v1.py:195]:
```suggestion
class FetchUserSiteSources:
```

File: eox_core/integrations/sentry.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ def load_class(full_class_string):
     return getattr(module, class_str)
 
 
-class ExceptionFilterSentry(object):
+class ExceptionFilterSentry():
```
Comment (morenol) [inline eox_core/integrations/sentry.py:29]:
```suggestion
class ExceptionFilterSentry:
```

File: eox_core/settings/test.py [hunk 5]
```diff
@@ -7,9 +7,8 @@
 from .common import *  # pylint: disable=wildcard-import, unused-wildcard-import
 
 
-class SettingsClass(object):
+class SettingsClass():
```
Comment (morenol) [inline eox_core/settings/test.py:10]:
```suggestion
class SettingsClass:
```

File: eox_core/storage.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
 EdxappDevelopmentStorage = get_edxapp_development_staticfiles_storage()  # pylint: disable=invalid-name
 
 
-class AbsoluteUrlAssetsMixin(object):
+class AbsoluteUrlAssetsMixin():
```
Comment (morenol) [inline eox_core/storage.py:15]:
```suggestion
class AbsoluteUrlAssetsMixin:
```

**Review by morenol — CHANGES_REQUESTED** [review]
Just a few linting suggestions, I think that we should adopt the usage of an autoformatting tool, for instance edX is [starting](https://discuss.openedx.org/t/transitioning-towards-autoformatting/3807) to use [black](https://github.com/psf/black). IT seems fine, what do you all think?

Could you add to the setup.py file a [python_requires argument](https://discuss.openedx.org/t/transitioning-towards-autoformatting/3807) to force the usage of python>=3.5, so users with Ironwood won't install this accidentally.

Also an entry in the Unreleased section of the Changelog would be fine

**Comment by magajh** [comment]
@morenol I think using an autoformatting tool would be a good idea, but black requires Python 3.6.0+ to run

**Comment by morenol** [comment]
> @morenol I think using an autoformatting tool would be a good idea, but black requires Python 3.6.0+ to run

@magajh  Yes, but I think that it is not a big problem, we can have a circle ci worker that checks that using python3.8 and internally we can install that using python3.6+.

File: package.json [hunk 7]
```diff
@@ -27,7 +27,7 @@
     "css-loader": "0.28.8",
     "extract-text-webpack-plugin": "2.1.2",
     "js-cookie": "2.2.0",
-    "node-sass": "4.7.2",
+    "node-sass": "4.14",
```
Comment (morenol) [inline package.json:30]: I think that we need to add the change to the package-lock.json file after this change

Comment (magajh) [reply package.json:30]: Done

File: tox.ini [hunk 8]
```diff
@@ -1,16 +1,15 @@
 [tox]
-envlist = py27-django111,py35-django{111,22}
+envlist = py35-django22-stretch,py38-django22-buster
```
Comment (morenol) [inline tox.ini:2]:
```suggestion
envlist = py{35,38}-django22
```

Comment (magajh) [reply tox.ini]: Done

**Review by morenol — APPROVED** [review]

**Pull request merged**